mri/operators/linear/wavelet.py

```diff
@@ -57,60 +67,97 @@ def get_coeff(self):
     def set_coeff(self, coeffs):
         self.transform.analysis_data = coeffs
 
+    def _op(self, data):
+        if isinstance(data, np.ndarray):
+            data = pysap.Image(data=data)
+        self.transform.data = data
+        self.transform.analysis()
+        coeffs, coeffs_shape = flatten(self.transform.analysis_data)
+        return coeffs, coeffs_shape
+
     def op(self, data):
         """ Define the wavelet operator.
-
         This method returns the input data convolved with the wavelet filter.
-
         Parameters
         ----------
         data: ndarray or Image
             input 2D data array.
-
         Returns
         -------
         coeffs: ndarray
             the wavelet coefficients.
         """
-        if isinstance(data, np.ndarray):
-            data = pysap.Image(data=data)
-        self.transform.data = data
-        self.transform.analysis()
-        coeffs, self.coeffs_shape = flatten(self.transform.analysis_data)
+        if self.n_coils > 1:
+            coeffs, self.coeffs_shape = zip(
+                *Parallel(
+                    n_jobs=self.n_jobs,
+                    backend=self.backend,
+                    verbose=self.verbose
+                )(
+                    delayed(self._op)
+                    (data[i])
```

Comment:
do the function and the arguments have to be on separate lines?

Reply:
Nope, Well I just found it easier to read this way..

Reply:
ok this is going to be subjective so let's leave it at that